openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -285,6 +286,9 @@ def test_home_page(
     ):
         self.create_user_for_course(self.course, user_type)
 
+        # Set course to invite-only
+        self.course.invitation_only = True
+
```

Comment:
You can check for the new message text (shown when user is unenrolled and the course is invitation_only) here.

Reply:
Will I not need the response from self.client?

Reply:
Sorry for the confusion. I meant you can add the check in this test. You will need the response from self.client for it.

Reply:
I could not get the test for this to pass. Looking further into the code, it looks like self.course is not used directly in the underlying functions. It's resolved from the course key. As a result, setting invitation_only here is not really visible in the other parts of the code. The only way seems to be actually creating a private course. But dont see a way of doing this without changing the test input data. We can either change the input(which is what I had previously, or write another test function?)

Reply:
Sorry, if I change the error message you proposed to "'You must be enrolled in the course to see course content.'" the tests pass. Is that ok?

Reply:
Sure, that should be fine too.